### PR TITLE
AFHTTPSessionManager copying securityPolicy not getting copied

### PR DESCRIPTION
**Pull request by kcharwood**
When I tried to copy the AFHTTPSessionManager the securityPolicy doesn't get copied with it.

### CODE REVIEW

**Comment by kcharwood** [comment]
cc @cnoon @kylef @jshier for a sanity check here.

Also updated `AFHTTPSessionManager` and `AFHTTPRequestManager` to use the copying/encoding/decoding features for the security policy as well.

This will need to go into 3.0.0 as well.

File: AFNetworking/AFSecurityPolicy.m [hunk 1]
```diff
@@ -308,4 +308,44 @@ + (NSSet *)keyPathsForValuesAffectingPinnedPublicKeys {
     return [NSSet setWithObject:@"pinnedCertificates"];
 }
 
+#pragma mark - NSSecureCoding
+
++ (BOOL)supportsSecureCoding {
+    return YES;
+}
+
+- (id)initWithCoder:(NSCoder *)decoder {
```
Comment (kylef) [inline AFNetworking/AFSecurityPolicy.m]: Can you change this `id` to `instancetype` please.

**Comment by kylef** [comment]
:+1: from me.

File: AFNetworking/AFSecurityPolicy.m (hunk 1 above)
Comment (kcharwood) [inline AFNetworking/AFSecurityPolicy.m]: Doh!. Updated.

**Comment by kcharwood** [comment]
Merged in to the 3_0_0 branch with f573fdb

**Pull request merged**